template.yml

```diff
@@ -0,0 +1,221 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Transform: AWS::Serverless-2016-10-31
+
+Mappings:
+  Constants:
+    DatadogCfnMacro:
+      Version: 0.1.0
+
+Parameters:
+  FunctionName:
+    Type: String
+    Default: DatadogCfnMacroLambda
+    Description: The name of the Lambda function that is invoked when the macro is run.
+  LogRetentionInDays:
+    Type: Number
+    Default: 90
+    Description: CloudWatch log retention for logs generated by the Datadog CloudFormation Macro Lambda function
+  SourceZipUrl:
+    Type: String
+    Default: ""
+    Description: DO NOT CHANGE unless you know what you are doing. Override the default location of the function source code.
+  PermissionsBoundaryArn:
+    Type: String
+    Default: ""
+    Description: ARN for the Permissions Boundary Policy
+
+Conditions:
+  IsAWSChina:
+    Fn::Equals:
+      - Ref: AWS::Partition
+      - "aws-cn"
+  SetFunctionName:
+    Fn::Not:
+      - Fn::Equals:
+          - Ref: FunctionName
+          - "DatadogCfnMacroLambda"
+  SetSourceZipUrl:
+    Fn::Not:
+      - Fn::Equals:
+          - Ref: SourceZipUrl
+          - ""
+  SetPermissionsBoundary:
+    Fn::Not:
+      - Fn::Equals:
+          - Ref: PermissionsBoundaryArn
+          - ""
+  SetS3SourceZip:
+    Fn::Equals:
+      - !Select [0, !Split ["/", !Ref SourceZipUrl]]
+      - "s3:"
+
+Resources:
+  MacroFunction:
+    Type: AWS::Serverless::Function
+    DependsOn: MacroFunctionZip
+    Properties:
+      FunctionName:
+        Fn::If:
+          - SetFunctionName
+          - Ref: FunctionName
+          - Ref: AWS::NoValue
+      Description: Processes a CloudFormation template to install Datadog Lambda layers for Python and Node.js Lambda functions.
+      Handler: index.handler
+      Runtime: nodejs12.x
+      CodeUri:
+        Bucket: !Ref MacroFunctionZipsBucket
+        Key:
+          Fn::Sub:
+            - "dd-cfn-macro-${Version}.zip"
+            - { Version: !FindInMap [Constants, DatadogCfnMacro, Version] }
+      Tags:
+        dd_cfn_macro_version: !FindInMap [Constants, DatadogCfnMacro, Version]
+      Policies:
+        - Version: "2012-10-17"
+          Statement:
+            - Effect: Allow
+              Action:
+                - logs:DescribeLogGroups
+                - logs:DescribeSubscriptionFilters
+              Resource: "*"
+  MacroFunctionLogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName:
+        Fn::Sub: /aws/lambda/${MacroFunction}
+      RetentionInDays:
+        Ref: LogRetentionInDays
```

Comment:
Not quite sure if it's necessary to have a log group for the macro lambda function, but I think it would help with debugging if customers are running into issues?